### DISCUSSION TITLE
The Energy dashboard versus Octopus's bills

### DISCUSSION THREAD

**Discussion by MHKnowles** [BottlecapDave/HomeAssistant-OctopusEnergy#989] [General]
Hi all,

Whilst I have the energy dashboard working in HA and it appears to bear a rough approximation to the bills I receive from Octopus for the same period, can others please share the entities they use if they're able to get it bang on? (or is that not possible?)

**Comment by BottlecapDave** [comment]
Hello and sorry for the very late response. Assuming you've followed the [guide](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/energy_dashboard/) for setting up the energy dashboard, then your bill will be represented by the previous consumption sensors. These sensors represent the values reported by your smart meter and are used for billing purposes. 

The consumption data should be identical to what appears on your bill, assuming HA was successful in pulling down all days worth of data. 

The costings however are a little bit more complicated. The costings are calculated using the rate information for the period, including VAT. This is slightly different in the breakdowns you might find in your bill on a per day basis which excludes VAT. If you look at each day in the energy dashboard and subtract the VAT you should get the value on your bill for that day. 

The totals found in the energy dashboard also don't include the standing charges (and associated VAT). This is because the energy dashboard stores the data on a half hourly basis and there is no where to associate the standing charge. I could spread the standing charge across all records, but then this would be wrong if you looked at a few hours. Therefore if you take your total on your bill and subtract your standing charge (no of days * standing charge rate + VAT) you should get to the total displayed in the energy dashboard (+/- a penny for rounding issues in HA)
